modules/lti-service-impl/src/main/java/org/opencastproject/lti/service/impl/LtiServiceImpl.java

```diff
@@ -246,60 +250,84 @@ public void upsertEvent(
-
-      final EventCatalogUIAdapter adapter = getEventCatalogUIAdapter();
+      r.setMediaPackage(mp);
+      final MetadataList metadataList = new MetadataList();
+      final MediaPackageElementFlavor flavor = new MediaPackageElementFlavor("dublincore", "episode");
+      final EventCatalogUIAdapter adapter = catalogUIAdapters.stream()
+          .filter(e -> e.getFlavor().equals(flavor)).findAny().orElse(null);
+      if (adapter == null) {
+        throw new RuntimeException("no adapter found");
+      }
       final DublinCoreMetadataCollection collection = adapter.getRawFields();
 
+      r.setMediaPackage(ingestService.addTrack(file.getStream(), file.getSourceName(),
+          MediaPackageElements.PRESENTER_SOURCE, mp));
+
+      JSONArray metadataJsonArray = (JSONArray) new JSONParser().parse(metadataJson);
+
       JSONArray collectionJsonArray = MetadataJson.extractSingleCollectionfromListJson(metadataJsonArray);
       MetadataJson.fillCollectionFromJson(collection, collectionJsonArray);
 
       replaceField(collection, "isPartOf", seriesId);
+      replaceField(collection, "duration", "6000");
       adapter.storeFields(mp, collection);
 
       AccessControlList accessControlList = null;
 
-      // If series is set and it's ACL is not empty, use series' ACL as default
-      if (StringUtils.isNotBlank(seriesId)) {
-        accessControlList = seriesService.getSeriesAccessControl(seriesId);
-      }
-
-      if (accessControlList == null || accessControlList.getEntries().isEmpty()) {
-        accessControlList = new AccessControlList(
+      accessControlList = new AccessControlList(
           new AccessControlEntry("ROLE_ADMIN", "write", true),
           new AccessControlEntry("ROLE_ADMIN", "read", true),
           new AccessControlEntry("ROLE_OAUTH_USER", "write", true),
           new AccessControlEntry("ROLE_OAUTH_USER", "read", true));
-      }
 
-      this.authorizationService.setAcl(mp, AclScope.Episode, accessControlList);
-      mp = ingestService.addTrack(file.getStream(), file.getSourceName(), MediaPackageElements.PRESENTER_SOURCE, mp);
+      r.setAcl(accessControlList);
+      r.setProcessing(
+              (JSONObject) new JSONParser().parse(
+                      String.format("{\"workflow\":\"%s\",\"configuration\":%s}", workflow, workflowConfiguration)));
+      r.setMetadataList(metadataList);
+      metadataList.add(adapter, collection);
 
-      final Map<String, String> configuration = gson.fromJson(workflowConfiguration, Map.class);
-      configuration.put("workflowDefinitionId", workflow);
-      ingestService.ingest(mp, workflow, configuration);
+      JSONObject source = new JSONObject();
+      source.put("type", "UPLOAD");
+      r.setSource(source);
+      indexService.createEvent(r);
+    } catch (SchedulerException e) {
+      if (e.getCause() != null && e.getCause() instanceof NotFoundException
+              || e.getCause() instanceof IllegalArgumentException) {
```

Comment:
`null instanceof ClassX` is always `false` anyway
```suggestion
      if (e.getCause() instanceof NotFoundException || e.getCause() instanceof IllegalArgumentException) {
```